src/status.rs

```diff
@@ -191,8 +203,58 @@ fn get_heartbeats(heartbeats: Option<Vec<HeartbeatValidatorMessage>>) -> Vec<Hea
     }
 }
 
-fn is_initializing() -> bool {
-    todo!()
+fn is_hb_by_validator(validator: &ValidatorDesc, heartbeat: &HeartbeatValidatorMessage) -> bool {
+    heartbeat.from == validator.id
+}
+
+fn get_new_state(last_approved: Option<LastApproved>) -> Vec<NewState> {
+    match last_approved {
+        Some(last_approved) => last_approved.new_state
+            .into_iter()
+            .filter_map(|new_state| match new_state.msg {
+                MessageTypes::NewState(new_state) => Some(new_state),
+                _ => None,
+            })
+            .collect(),
+        None => Default::default(),
+    }
+}
+
+fn get_approve_state(last_approved: Option<LastApproved>) -> Vec<ApproveState> {
```

Comment:
This along with get_new_state() and get_heartbeats() are quite similar. Could be merged into one function.